File: JUnitProject/src/test/java/base/BaseTest.java
package base;

import factory.WebDriverFactory;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.BeforeEach;
import org.openqa.selenium.WebDriver;

public abstract class BaseTest {

    protected WebDriver webDriver;
    private String url = "https://testfaceclub.com/login-employee-v2/";

    public BaseTest(){
        System.out.println("BaseTest Thread-id: " + Thread.currentThread().getId());
    }

    @BeforeAll
    public static void setUpBaseTest(){
        System.out.println("setUpBaseTest Thread-id: " + Thread.currentThread().getId());
    }

    @BeforeEach
    public void setUp() throws Exception {
        System.out.println("setUp Thread-id: " + Thread.currentThread().getId());

        webDriver = WebDriverFactory.getDriver("chrome", "");
        webDriver.get(url);
        webDriver.manage().window().maximize();
    }

    @AfterEach
    public void tearDown(){
        if(webDriver != null){
            webDriver.quit();
        }
    }
}

File: NUnitProject/NUnitProject/Tests/Base/BaseTest.cs
using NUnit.Framework;
using NUnitProject.FactoryWebDriver;
using OpenQA.Selenium;
using System;
using System.Configuration;
using System.Diagnostics;
using System.Threading;

namespace NUnitProject.Tests.Base
{
    public abstract class BaseTest
    {
        [ThreadStatic]
        protected static IWebDriver WebDriver;

        private String Url = "https://testfaceclub.com/login-employee-v2/";

        public BaseTest()
        {
            Logger.Instance.Info("BaseTest Thread-id: " + Thread.CurrentThread.ManagedThreadId);
        }

        [SetUp]
        public void SetUp()
        {
            Logger.Instance.Info("setUp Thread-id: " + Thread.CurrentThread.ManagedThreadId);

            WebDriver = WebDriverFactory.GetDriver(ConfigurationManager.AppSettings.Get("browser"),
                                                    ConfigurationManager.AppSettings.Get("seleniumGridUrl"));
            WebDriver.Navigate().GoToUrl(Url);
            WebDriver.Manage().Window.Maximize();
        }

        [TearDown]
        public void TearDown()
        {
            if (WebDriver != null)
            {
                WebDriver.Quit();
            }
        }
    }
}
